vpr/src/device/rr_graph_builder.h

```diff
@@ -0,0 +1,63 @@
+#ifndef RR_GRAPH_BUILDER_H
+#define RR_GRAPH_BUILDER_H
+
+#include "rr_graph_storage.h"
+#include "rr_spatial_lookup.h"
+
+/* A data structure allows data modification on a routing resource graph 
+ *
+ * Note that the builder does not own the storage
+ * It serves a virtual protocol for
+ * - node_storage: store the node list 
+ * - node_lookup: store a fast look-up for the nodes
+ *
+ * Note:
+ * - This is the only data structre allowed to modify a routing resource graph
+ *
+ */
+class RRGraphBuilder {
+    /****************
+     * Constructors
+     ****************/
+  public:
+    RRGraphBuilder(t_rr_graph_storage* node_storage,
```

Comment:
Might be useful to add a comment about what the parameters are (even if it changes in the future, but as reference for the reader).

Reply:
Yes. Since we will comment each private data structure, I suggest just simply redirect developers to the comments in the internal data storage section of the header file.